Unify action buttons of lists

8191 opened the pull request:
* Changed "X" icon for delete action to "trash bin" for all delete actions, which modify the config (e.g. remove firewall rule). This means, e.g. removing a DHCP lease or a state from the pf state table still has an "X" and not a "trash bin" icon. #616

* Moved position of most tooltips from left (most) to top. Not moved were tooltips of horizontal button rows (e.g. user manager: add/remove groups) and the services table. #616 

_Unfortunately this PR depends on #626._

fichtner commented:
please don't base PRs on other PRs, it creates really messy histories. I will cherry-pick the commits instead here for clarity

AdSchellevis commented:
oops, I placed some comments in the other commit, but to be sure. can we move get_alias_description to legacy_bindings.inc and skip the is_alias there?

fichtner commented:
The stacked PRs do not even play nice with github. this one should go in first, the others need to be rebased afterwards.

fichtner commented:
Merged manually. Please rebase your other PR :)

8191 commented:
@AdSchellevis I'll move `get_alias_description` to legacy_bindings.inc, but what do you mean by _skip is_alias_?

AdSchellevis commented:
@8191 thanks, the is_alias checks the global $aliastable for key existence, which in return is populated by the same data that the get_alias_description is traversing. 
If performance is a bottleneck (which I don't expect) it might be a good idea to combine actions somehow (build a class which keeps it's own reference in stead of trusting application globals).
So my strategy would be to first try the code without caching, then if needed solve those.
For profiling, I use webgrind from time to time which can easily be installed on OPNsense.
Eventually I would like to kill all globals (except $config, which is too tangled into the codebase).

8191 commented:
@AdSchellevis Thanks for explanation, I just found your post in the other commit comment. ;)
I've removed the `is_alias` call and moved the function to legacy_bindings.inc.
